### PR TITLE
enable rerun "deck-private"

### PR DESCRIPTION
**Pull request by clarketm**
Still need to:
- [x] create the Oauth app, 
- [x] install the secrets, 
- [x] and add support for configuring the `rerun_auth_config` by Github `repo` or `org` ~https://github.com/kubernetes/test-infra/pull/15393~, ~https://github.com/kubernetes/test-infra/pull/15396~, https://github.com/kubernetes/test-infra/pull/15437

https://github.com/istio/test-infra/blob/8d4d38afcf205ff6e100790f5f7f2a814037c1cc/prow/config.yaml#L47-L55

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Comment by clarketm** [comment]
- [x] Still waiting on upstream feature to be reviewed: kubernetes/test-infra#15437

**Comment by clarketm** [comment]
/hold
- [x] until Prow is bumped to the correct version (>= kubernetes/test-infra@d7464ed97).

**Comment by clarketm** [comment]
New prow (i.e. kubernetes/test-infra@e079f1a17) deployed. This is now ready.

File: prow/cluster/deck_private_deployment.yaml [hunk 1]
```diff
@@ -30,10 +30,21 @@ spec:
         - --redirect-http-to=prow-private.istio.io
         - --kubeconfig=/etc/kubeconfig/istio-config
         - --gcs-credentials-file=/etc/service-account/service-account.json
-        - --github-token-path=/etc/github/oauth
         - --spyglass
         - --hidden-only
+        - --rerun-creates-job
+        - --oauth-url=/github-login
+        - --github-token-path=/etc/github/oauth
+        - --github-endpoint=http://ghproxy
```
Comment (cjwagner) [inline prow/cluster/deck_private_deployment.yaml:38]: This flag should be used twice to specify the upstream API URL as a fallback for when ghproxy in unavailable: https://github.com/istio/test-infra/search?q=--github-endpoint&type=Code

Comment (clarketm) [reply prow/cluster/deck_private_deployment.yaml:38]: Missed that, thanks. Updated

**Review by cjwagner — APPROVED** [review]

**Pull request merged**